src/main/java/io/reactivex/Scheduler.java

```diff
@@ -106,13 +106,10 @@ public Disposable schedule(Runnable run) {
         }
         
         public Disposable schedulePeriodically(Runnable run, final long initialDelay, final long period, final TimeUnit unit) {
-            final SerialDisposable first = new SerialDisposable();
-
-            final SerialDisposable sd = new SerialDisposable(first);
-            
             final Runnable decoratedRun = RxJavaPlugins.onSchedule(run);
-            
-            first.replace(schedule(new Runnable() {
+
+            final SerialDisposable sd = new SerialDisposable();
```

Comment:
But `sd` isn't returned for the calling code to even call `dispose()` on before we replace its value with the `Disposable` return from delegating to `schedule`.